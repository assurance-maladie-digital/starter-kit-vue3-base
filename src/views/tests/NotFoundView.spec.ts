import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import NotFound from '../NotFoundView.vue'
import { vuetify } from '../../../tests/unit/setup'

describe('NotFoundView', () => {
    it('renders properly', () => {
        const wrapper = mount(NotFound, {
            global: {
                plugins: [
                    vuetify,
                ],
            }
        })
        expect(wrapper.html()).toMatchSnapshot();

        const h2Text = wrapper.find('h2').text();
        expect(h2Text).toContain('Page non trouv√©e');
    })
})
