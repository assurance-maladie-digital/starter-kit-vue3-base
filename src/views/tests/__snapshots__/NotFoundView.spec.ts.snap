// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFoundView > renders properly 1`] = `
"<div data-v-813c93fa="" data-v-4104c579="" data-v-a0c2a46f="" class="py-10 px-8 vd-page-container d-flex justify-center mb-6">
  <div data-v-813c93fa="" class="v-sheet v-theme--light bg-transparent" style="width: 800px;">
    <div data-v-4104c579="" class="v-card v-theme--light v-card--density-default elevation-2 v-card--variant-elevated pa-6 pa-sm-16">
      <!---->
      <div class="v-card__loader">
        <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
          <!---->
          <div class="v-progress-linear__background"></div>
          <div class="v-progress-linear__buffer" style="width: 0%;"></div>
          <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
            <div class="v-progress-linear__indeterminate">
              <div class="v-progress-linear__indeterminate long"></div>
              <div class="v-progress-linear__indeterminate short"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <!---->
      <div data-v-4104c579="" class="v-row max-width-none">
        <div data-v-4104c579="" class="v-col-sm-6 v-col-12 order-last order-sm-first text-center text-sm-left">
          <div data-v-4104c579="" class="vd-code font-weight-thin text-primary mb-4"><span data-v-4104c579="" class="d-sr-only">Code d'erreur&nbsp;: </span> 404</div>
          <h2 data-v-4104c579="" class="mb-2 font-weight-bold text-h5 mb-4">Page non trouvée</h2>
          <p data-v-4104c579="">Cette page n’existe pas ou a été déplacée.</p><a data-v-4104c579="" class="v-btn v-btn--elevated v-theme--light bg-primary v-btn--density-default v-btn--size-default v-btn--variant-elevated mt-6" href="/home"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator="">Retour à l’accueil</span>
            <!---->
            <!---->
          </a>
        </div>
        <div data-v-4104c579="" class="v-col-sm-6 v-col-12 d-flex align-center justify-center"><img data-v-a0c2a46f="" src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20fill='none'%20viewBox='0%200%20175%20185'%3e%3cpath%20fill='%23e7ecf5'%20fill-rule='evenodd'%20d='M82.05%201.03c25.02%204.23%2035.74%2031.89%2052.64%2050.76%2015.22%2017%2038.8%2029.7%2039.77%2052.46.99%2023.2-15.89%2044.01-35.1%2057.13-16.47%2011.25-37.51%203.39-57.3%205.99-25.97%203.41-55.68%2029.28-74.73%2011.37-18.93-17.8%207.49-48.54%207.6-74.49.11-25.7-20.98-52.27-6.95-73.83C23%207.3%2054.8-3.56%2082.05%201.03Z'%20clip-rule='evenodd'/%3e%3cpath%20fill='%23fff'%20d='M101.76%20132.91%2080.07%2096.9l-6.74-11.21%203.21-4.8s32.94-.8%2035.35%200l35.34%2056.04-4.01%204.8h-36.15l-5.31-8.82Z'/%3e%3cpath%20fill='%230c419a'%20d='m101.76%20132.91%205.3%208.83h-.4l-52.48-8.01%2047.58-.82Z'/%3e%3cpath%20fill='%23fff'%20d='M54.18%20133.73h-.13V96.9h26.03l21.68%2036.01-47.58.82ZM54.05%2096.9v36.83l-19.28-9.6V88.9l19.28%208Z'/%3e%3cpath%20fill='%230c419a'%20d='m63.87%2064.26%2017.25-5.12%2020.51-6.1s2.54%204.43%209.87%202.6c-5.53%201.83-20.28%206.68-26.74%208.62-8.04%202.4-20.89%200-20.89%200Z'/%3e%3cpath%20fill='%23fff'%20d='M111.5%2055.64c-7.34%201.83-9.87-2.6-9.87-2.6l-20.51%206.1c1.6-1.37%207.76-7.03%208.46-13.3.8-7.2%203.21-15.21%209.64-18.41%205.82-2.9%2027.96-15.14%2032.1-17.43-1.19.84-2.39%202.33-3.42%204.89a28.8%2028.8%200%200%200-1.37%204.53c-3.21%2014.41-2.4%2031.23-12.05%2035.23-1.07.44-2.06.77-2.97%201Z'/%3e%3cpath%20fill='%230c419a'%20d='M131.32%2010c2.88-2.03%205.66-.19%205.66-.19L128%2014.93l-.1-.04c1.03-2.56%202.23-4.06%203.42-4.9Z'/%3e%3cpath%20stroke='%230c419a'%20stroke-linecap='round'%20stroke-linejoin='round'%20stroke-miterlimit='10'%20stroke-width='2'%20d='m111.5%2055.64%202.14-.7M111.5%2055.65c-7.33%201.82-9.87-2.6-9.87-2.6l-20.51%206.09-17.25%205.12s12.85%202.4%2020.89%200c6.46-1.94%2021.2-6.79%2026.74-8.61ZM111.5%2055.65h.01M128%2014.93l8.98-5.12s-2.78-1.84-5.66.19c-1.19.84-2.39%202.33-3.42%204.89a28.8%2028.8%200%200%200-1.37%204.53c-3.21%2014.41-2.4%2031.23-12.05%2035.23-1.07.45-2.06.77-2.97%201'/%3e%3cpath%20stroke='%230c419a'%20stroke-linecap='round'%20stroke-linejoin='round'%20stroke-miterlimit='10'%20stroke-width='2'%20d='M80.74%2059.45s.14-.1.38-.31c1.6-1.37%207.76-7.03%208.46-13.3.8-7.2%203.21-15.21%209.64-18.41%205.82-2.9%2027.96-15.14%2032.1-17.43l.67-.37M97.61%2033.03l20.08-10.4M96%2040.24l20.9-10.41M62.08%2088.9H34.77l19.28%208h26.1M101.79%20132.91h-.03l-47.58.82h-.13V96.9M54.05%20133.73l-19.28-9.6V88.9M76.54%2080.89l34.54%2056.04h26.52'/%3e%3cg%20stroke='%230c419a'%20stroke-linecap='round'%20stroke-linejoin='round'%20stroke-width='2'%3e%3cpath%20stroke='%230c419a'%20stroke-miterlimit='10'%20d='m54.06%20133.7.12.03%2052.49%208h36.55l4.01-4.8L111.9%2080.9c-2.41-.8-35.35%200-35.35%200l-3.22%204.8%206.75%2011.21%2021.69%2036.02%205.3%208.82'/%3e%3cpath%20d='M1-1h6.27'%20transform='rotate(81.49%20-7.48%2056.18)%20skewX(.05)'/%3e%3cpath%20d='M1-1h7.26'%20transform='rotate(48.66%20-44.44%2074.2)%20skewX(.16)'/%3e%3cpath%20d='M1-1h4.39'%20transform='matrix(.9%20.45%20-.45%20.9%2038.17%2066.32)'/%3e%3c/g%3e%3c/svg%3e" alt="Page non trouvée"></div>
      </div>
      <!---->
      <!----><span class="v-card__underlay"></span>
    </div>
  </div>
</div>
<div data-v-813c93fa="" class="pa-0 pb-8 vd-page-container d-flex justify-center">
  <div data-v-813c93fa="" class="v-sheet v-theme--light bg-transparent" style="width: 800px;"><button type="button" class="v-btn v-btn--elevated v-theme--light bg-primary v-btn--density-default v-btn--size-default v-btn--variant-elevated setSupportId"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator=""> Afficher l’identifiant de support </span>
      <!---->
      <!---->
    </button></div>
</div>"
`;
